modules/touch/src/touchinteraction.cpp

```diff
@@ -829,54 +853,31 @@ void TouchInteraction::computeVelocities(const std::vector<TouchInputHolder>& li
-                [&](double d, const TouchInputHolder& c) {
-                    const glm::vec2 currPos = { c.latestInput().x, c.latestInput().y };
-                    return d + glm::distance(currPos, _centroid);
-                }
-            ) / list.size();
-            double lastDistance = std::accumulate(
-                lastProcessed.begin(),
-                lastProcessed.end(),
-                0.f,
-                [&](float d, const TouchInput& p) {
-                    const glm::vec2 lastPos = { p.x, p.y };
-                    return d + glm::distance(lastPos, _centroid);
-                }
-            ) / lastProcessed.size();
-
-            glm::dvec3 camPos = _camera->positionVec3();
-            glm::dvec3 centerPos = anchor->worldPosition();
-            glm::dvec3 currDistanceToFocusNode = camPos - centerPos;
-
-            const double distanceFromFocusSurface =
-                length(currDistanceToFocusNode) - anchor->boundingSphere();
-            double zoomFactor = (distance - lastDistance);
+             // points this/first frame
+            using namespace glm;
+            const TouchInput& startFinger0 = _pinchInputs[0].firstInput();
+            const TouchInput& startFinger1 = _pinchInputs[1].firstInput();
+            const dvec2 startVec0 = dvec2(startFinger0.x * aspectRatio, startFinger0.y);
+            const dvec2 startVec1 = dvec2(startFinger1.x * aspectRatio, startFinger1.y);
+            double distToCentroidStart = length(startVec0 - startVec1) / 2.0;
+
+            const TouchInput& endFinger0 = _pinchInputs[0].latestInput();
+            const TouchInput& endFinger1 = _pinchInputs[1].latestInput();
+            const dvec2 endVec0 = dvec2(endFinger0.x * aspectRatio, endFinger0.y);
+            const dvec2 endVec1 = dvec2(endFinger1.x * aspectRatio, endFinger1.y);
+            double distToCentroidEnd = length(endVec0 - endVec1) / 2.0;
+
+            double zoomFactor = distToCentroidEnd - distToCentroidStart;
 #ifdef TOUCH_DEBUG_PROPERTIES
             pinchConsecCt++;
             pinchConsecZoomFactor += zoomFactor;
 #endif
 
-            _constTimeDecayCoeff.zoom = computeConstTimeDecayCoefficient(_vel.zoom);
-            if (distanceFromFocusSurface > 0.1) {
-                const double ratioOfDistanceToNodeVsSurface =
-                    length(currDistanceToFocusNode) / distanceFromFocusSurface;
-                if (ratioOfDistanceToNodeVsSurface > _zoomSensitivityDistanceThreshold) {
-                    zoomFactor *= pow(
-                        std::abs(distanceFromFocusSurface),
-                        static_cast<float>(_zoomSensitivityExponential)
-                    );
-                }
-            }
-            else {
-                zoomFactor = 1.0;
-            }
-            _vel.zoom = zoomFactor * _zoomSensitivityProportionalDist *
-                         std::max(_touchScreenSize.value() * 0.1, 1.0);
+            _constTimeDecayCoeff.zoom = 1.0;
```

Comment:
_constTimeDecayCoeff is this really a needed coeff if it is always set to 1?

Reply:
For continuous zoom I set this 'coeff' to 1, since i don't want the zoom to have any falloff while stepping the position. Once the user releases both fingers, the coeff is set to whatever is specified based on the current zoom velocity in the `resetAfterInput` function

Reply:
Wouldn't it be better to not include it in the calulations in this case instead of setting and resetting it back and forth? But if that is too complicated this solution also works.

Reply:
So, the touch-stuff has a `step` function which steps the camera with delta_time each frame based on velocities saved in some structure (orbit, zoom, roll, pan velocities). It also calls a `decelerate` function with this same delta_time. 

In this case, as a "pinch-gesture" happens, we don't want the zoom velocity to change until we release our fingers. To overcome this I set the decay-coeff of `zoom` to 1, which basically says "don't decelerate this zoom-field when you're decelerating all the velocity-stuff".

All the other velocities(such as orbit and potentially panning?) should still be decelerated, so that we get some friction-motion going on while spinning around, so I don't touch the other coefficients. 

When resetting the input (which happens when the user removes all fingers from the screen) we then set the decay-coeff for zoom, which tells the decelerate function to start decelerating the zoom-velocity. So I feel like it's the most straight-forward solution

Reply:
Ok fair enough, That is a good explanation. :)